diagnostic.md

```diff
@@ -6,70 +6,71 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+resources :books, only: [:show]
+resources :books, except: [:new, :edit]
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+This goes into the routes.rb file.
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+This command looks for migrations created by the user and runs the up code for any migrations that are in the down position.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+`rails g controller things` will only create an empty controller file for the things model.  `rails g scaffold things` will create a serializer, a controller, a model, routes, and will also populate the CRUD routes for the things controller.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+A serializer controls which attributes of the object to return in a GET request, in JSON format.  So far I have used the serializer to prevent the different timestamps from showing up in my requests so that my screen isn't cluttered by them.
 ```
 
 What do we expect the command `Patient.all` to return?
 
 ```md
-// your response here
+It will return a hash of all of the patient objects stored in the patients database.
 ```
 
 We use the constant `Patient` in the `PatientsController`, but where is it
 actually defined?
 
 ```md
-// your response here
+Patient is defined in the model file, patient.rb.
 ```
 
 Why do our `model` files appear to be lacking so much code, when they're doing
 so much for us?
 
 ```md
-// your response here
+Most of the things that are happening with the model is defined on ApplicationRecord, from which our models inherit.  Rails allows us to change and manipulate only the things most important to our program and not the bones that make up rails.
```

Comment:
`The methods making use of the Patient model are defined within the controller.`